bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1089 @@
+
+- They advertise `option_taproot_gossip` in the `init` or 
+  legacy `node_announcement` messages
+  OR
+- They have broadcast one of the new gossip messages defined in this document.
+
+Advertisement of the `option_taproot_gossip` feature bit should be taken to 
+mean:
+
+- The node understands and is able to forward all the taproot gossip messages.
+- The node may be willing to open an announced taproot channel.
+
+### `open_channel` Extra Requirements
+
+These extra requirements only apply if the `option_taproot` channel type is set
+in the `open_channel` message.
+
+The sender:
+
+- if `option_taproot_gossip` was negotiated:
+    - MAY set the `announce_channel` bit in `channel_flags`
+- otherwise:
+    - MUST NOT set the `announce_channel` bit.
+
+The receiving node MUST fail the channel if:
+
+- if `option_taproot_gossip` was not negotiated and the `announce_channel`
+  bit in the `channel_flags` was set.
+
+### `channel_ready` Extensions
+
+These extensions only apply if the `option_taproot` channel type was set in the
+`open_channel` message along with the `announce_channel` channel flag.
+
+1. `tlv_stream`: `channel_ready_tlvs`
+2. types:
+   1. type: 0 (`announcement_node_pubnonce`)
+   2. data:
+      * [`66*byte`: `public_nonce`]
+   3. type: 2 (`announcement_bitcoin_pubnonce`)
+   4. data:
+      * [`66*byte`: `public_nonce`]
+   
+### Requirements
+
+The sender: 
+
+  - MUST set the `announcement_node_pubnonce` and 
+    `announcement_bitcoin_pubnonce` tlv fields in `channel_ready`.
+  - SHOULD use the MuSig2 `NonceGen` algorithm to generate two unique secret 
+    nonces and use these to determine the corresponding public nonces: 
+    `announcement_node_pubnonce` and `announcement_bitcoin_pubnonce`.
+  
+The recipient: 
+
+  - MUST fail the channel if: 
+    - if the `announcement_node_pubnonce` or `announcement_bitcoin_pubnonce` tlv 
+      fields are missing.
+    - if the `announcement_node_pubnonce` and `announcement_bitcoin_pubnonce` 
+      are equal.
+    
+### Rationale
+
+The final signature included in the `channel_announcement_v2` message will be a
+single [BIP-340][bip-340] signature that needs to be valid for a public key
+derived by aggregating all the `node_ID` and `bitcoin_key` public keys. The
+signature for this key will thus be created by aggregating (via MuSig2) four
+partial signatures. One for each of the keys: `node_ID_1`, `node_ID_2`, 
+`bitcoin_key_1` and `bitcoin_key_2`. A nonce will need to be generated and
+exchanged for each of the keys and so each node will need to send the other node
+two nonces. The `announcement_node_nonce` is for `node_ID_x` and the 
+`announcement_bitcoin_nonce` is for `bitcoin_key_x`.
+
+Since the channel can only be announced once the `channel_ready` messages have 
+been exchanged and since it is generally preferred to keep nonce exchanges as 
+Just In Time as possible, the nonces are exchanged via the `channel_ready` 
```

Comment:
Can we include a citation for the Just In Time preference? Reading [BIP327](https://github.com/bitcoin/bips/blob/master/bip-0327.mediawiki#design), it seems to suggest in bullet 3 that it doesn't really matter how early you exchange them since it is valid to do so even before the exact signing participants have been determined.